src/System.IO.Pipelines/ReadableBuffer.cs

```diff
@@ -56,8 +59,8 @@ internal ReadableBuffer(ReadCursor start, ReadCursor end)
         {
             _start = start;
             _end = end;
-            start.TryGetBuffer(end, out _first);
-            _length = -1;
+            _length = start.GetLength(_end);
```

Comment:
`_length = start.GetLength(end);`? Skip the RAW hazard? Probably doesn't matter but [hazards](https://en.wikipedia.org/wiki/Hazard_(computer_architecture)#Read_after_write_.28RAW.29) keep me up at night